From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bastiaan van der Plaat <bastiaan.v.d.plaat@gmail.com>
Date: Sat, 10 Oct 2026 20:06:21 +0200
Subject: [PATCH] Write the assembly in unified syntax for LLVM

LLVM's assembler only takes the unified syntax of ldm with condition
codes and of the flag setting Thumb and inline assembly instructions.
---
 include/calico/nds/arm7/aes.h | 4 ++--
 source/nds/bios.s             | 8 ++++----
 source/nds/irq_handler.32.s   | 4 ++--
 3 files changed, 8 insertions(+), 8 deletions(-)

diff --git a/include/calico/nds/arm7/aes.h b/include/calico/nds/arm7/aes.h
index f4eda16..59bdeeb 100644
--- a/include/calico/nds/arm7/aes.h
+++ b/include/calico/nds/arm7/aes.h
@@ -85,8 +85,8 @@ MK_INLINE void aesCtrIncrementIv(AesBlock* iv, u32 value)
 {
 	__asm__ __volatile__ (
 		"adds   %0, %0, %4\n\t"
-		"addcss %1, %1, #1\n\t"
-		"addcss %2, %2, #1\n\t"
+		"addscs %1, %1, #1\n\t"
+		"addscs %2, %2, #1\n\t"
 		"addcs  %3, %3, #1\n\t"
 		: "+r"(iv->data[0]), "+r"(iv->data[1]), "+r"(iv->data[2]), "+r"(iv->data[3])
 		: "Ir"(value)
diff --git a/source/nds/bios.s b/source/nds/bios.s
index 8060e0a..f56952d 100644
--- a/source/nds/bios.s
+++ b/source/nds/bios.s
@@ -80,19 +80,19 @@ FUNC_START16 svcGetPitchTable
 
 	@ Account for bugged table offset (DSi BIOS workaround)
 	ldr  r1, =0x46a
-	sub  r0, r1
+	subs r0, r1
 
 1:	svc  0x1b
 
 	@ TODO: is casting to u16 really necessary?
-	lsl  r0, #16
-	lsr  r0, #16
+	lsls r0, #16
+	lsrs r0, #16
 
 	bx   lr
 FUNC_END
 
 FUNC_START16 svcCustomHalt
-	mov  r2, r0
+	movs r2, r0
 	svc  0x1f
 	bx   lr
 FUNC_END
diff --git a/source/nds/irq_handler.32.s b/source/nds/irq_handler.32.s
index 03c2a4a..1b6bfff 100644
--- a/source/nds/irq_handler.32.s
+++ b/source/nds/irq_handler.32.s
@@ -23,7 +23,7 @@ FUNC_START32 __irq_handler
 	ldm   r12, {r0, r1}       @ read both IE and IF at the same time
 	ands  r0, r0, r1          @ r0 <- IO_IE & IO_IF
 #if defined(ARM9)
-	ldmeqia sp!, {r0-r3,r12,pc}^ @ GAS doesn't like pop {reglist}^; so use the "real" opcode instead
+	ldmiaeq sp!, {r0-r3,r12,pc}^ @ GAS doesn't like pop {reglist}^; so use the "real" opcode instead
 #elif defined(ARM7)
 	bne   1f
 
@@ -127,7 +127,7 @@ FUNC_START32 __irq_handler
 #if defined(ARM7)
 	ldreq pc, [sp, #-4] @ Return to BIOS if not
 #elif defined(ARM9)
-	ldmeqia sp!, {r0-r3,r12,pc}^
+	ldmiaeq sp!, {r0-r3,r12,pc}^
 #endif
 
 	@ s_curThread = s_deferredThread; s_deferredThread = NULL;
